core/src/main/scala/org/apache/spark/rpc/RpcEndpointAddress.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.rpc
+
+import org.apache.spark.SparkException
+
+/**
+ * An address identifier for an RPC endpoint.
+ *
+ * The `rpcAddress` may be null, in which case the endpoint is registered via a client-only
+ * connection and can only be reached via the client that sent the endpoint reference.
+ *
+ * @param rpcAddress The socket address of the endpoint.
+ * @param name Name of the endpoint.
+ */
+private[spark] case class RpcEndpointAddress(val rpcAddress: RpcAddress, val name: String) {
```

Comment:
I think this whole file shouldn't be here. Note that there's been a lot of shuffling around this class. In 1.6.0 there are two RPC implementations: Netty-based and Akka-based. In 2.0 the Akka one is removed.

This class already exists in your fork under `org.apache.spark.rpc.netty`. You should probably modify the heartbeat logic to work with both RPC implementations and remove this copy. This commit is related: 4f5a24d